src/simulation.hpp

```diff
@@ -447,16 +449,22 @@ template <typename problem_t> class AMRSimulation : public amrex::AmrCore
 #ifdef AMREX_PARTICLES
 	void InitParticles();	 // create tracer particles
 	void InitCICParticles(); // create CIC particles
+	void InitPhyParticles(); // create PhysicsParticles
 	int do_tracers = 0;
 	int do_cic_particles = 0;
+	int do_rad_particles = 0;
 	std::unique_ptr<amrex::AmrTracerParticleContainer> TracerPC;
 	std::unique_ptr<quokka::CICParticleContainer> CICParticles;
+	std::unique_ptr<quokka::RadParticleContainer<problem_t>> RadParticles;
 #endif
 
 	// external objects
 #ifdef AMREX_USE_ASCENT
 	Ascent ascent_;
 #endif
+
+	// Add PhysicsParticleRegister member
+	std::unique_ptr<quokka::PhysicsParticleRegister<problem_t>> particleRegister_;
```

Comment:
This does not need to be a pointer. It can just be `quokka::PhysicsParticleRegister<problem_t>`.